reporting/file_not_found_report.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+'''****************************************************************************
+FILE: file_not_found_report.py
+
+PURPOSE: Produce a report of requests with a 404 return code from an Apache log
+file. Shows number of occurrences per email, and optionally per day as well.
+
+PROJECT: Land Satellites Data System Science Research and Development (LSRD)
+    at the USGS EROS
+
+LICENSE TYPE: NASA Open Source Agreement Version 1.3
+
+AUTHOR: ngenetzky@usgs.gov
+
+****************************************************************************'''
+import sys
+import argparse
+import mapreduce_logfile as mapred
+
+
+def parse_arguments():
+    desc = ('Produce a report of requests with a 404 return code from an'
+            ' Apache log file. Shows number of occurrences per email, and'
+            ' optionally per day as well.')
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument('--show_per_day', dest='show_per_day',
+                        action='store_true', default=False,
+                        help='List offenses per day?')
+    args = parser.parse_args()
+    return args
+
+
+def main(iterable, show_per_day=False):
+    if(show_per_day):
+        return mapred.report_404_perdate_peremail_on_production_orders(iterable)
```

Comment:
im referring to your statement regarding namedtuples

On Mon, Aug 17, 2015 at 1:59 PM, Nathan Genetzky <notifications@github.com>
wrote:

> In reporting/file_not_found_report.py
> <https://github.com/USGS-EROS/espa-maintenance/pull/1#discussion_r37223692>
> :
>
> > +
> > +def parse_arguments():
> > +    desc = ('Produce a report of requests with a 404 return code from an'
> > +            ' Apache log file. Shows number of occurrences per email, and'
> > +            ' optionally per day as well.')
> > +    parser = argparse.ArgumentParser(description=desc)
> > +    parser.add_argument('--show_per_day', dest='show_per_day',
> > +                        action='store_true', default=False,
> > +                        help='List offenses per day?')
> > +    args = parser.parse_args()
> > +    return args
> > +
> > +
> > +def main(iterable, show_per_day=False):
> > +    if(show_per_day):
> > +        return mapred.report_404_perdate_peremail_on_production_orders(iterable)
>
> Are you referring to "report_404_perdate_peremail_on_production_orders" or
> "main"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/USGS-EROS/espa-maintenance/pull/1/files#r37223692>.
>



-- 
David V. Hill
LSRD
SGT
Contractor to USGS Earth Resources Observation And Science Center (EROS)
47914 252nd Street
Sioux Falls, SD 57198-9801
605.594.2639
dhill@usgs.gov